node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/MaintenanceDeployment.java

```diff
@@ -161,39 +118,37 @@ public boolean execute(boolean verifyTarget,
                                Agent agent, Deployer deployer, Metric metric, NodeRepository nodeRepository) {
             if (isEmpty()) return false;
             ApplicationId application = node.allocation().get().owner();
-            try (MaintenanceDeployment deployment = new MaintenanceDeployment(application, deployer, metric, nodeRepository)) {
-                if ( ! deployment.isValid()) return false;
-
-                boolean couldMarkRetiredNow = markPreferToRetire(node, true, agent, nodeRepository);
-                if ( ! couldMarkRetiredNow) return false;
-
-                Optional<Node> expectedNewNode = Optional.empty();
-                try {
-                    if ( ! deployment.prepare()) return false;
-                    if (verifyTarget) {
-                        expectedNewNode =
-                                nodeRepository.nodes().list(Node.State.reserved)
-                                              .owner(application)
-                                              .cluster(node.allocation().get().membership().cluster().id())
-                                              .except(node)
-                                              .first();
-                        if (expectedNewNode.isEmpty()) return false;
-                        if (!expectedNewNode.get().hasParent(toHost.hostname())) return false;
-                    }
-                    if ( deployment.activate().isEmpty()) return false;
-
-                    log.info(agent + " redeployed " + application + " to " +
-                             ( verifyTarget ? this : "move " + (node + " from " + fromHost.hostname())));
-                    return true;
+            MaintenanceDeployment deployment = new MaintenanceDeployment(application, deployer, metric, nodeRepository);
```

Comment:
Are you sure we don't need to take the lock in this method?

Reply:
I suppose since we have multiple node movers, there's a higher chance their deployment's prepare+active interleaving. Maybe we should have a separate node mover lock?

Reply:
Yes, that sounds like a good idea.